DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1

```diff
@@ -709,4 +813,40 @@ function Update-LocationTagInApplicationHostConfigForAuthentication
     $webAdminSrvMgr.CommitChanges()
 }
 
+function Find-CertificateThumbprintWithSubjectAndTemplateName
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [String]
+        $Subject,
+
+        [Parameter(Mandatory)]
+        [String]
+        $TemplateName,
+
+        [String]
+        $Store = 'Cert:\LocalMachine\My'
+    )
+
+    [Array] $CertificatesFromTemplates = (Get-ChildItem -Path $Store).Where{$_.Extensions.Oid.Value -contains '1.3.6.1.4.1.311.20.2'}
+
+    $Certificate = $CertificatesFromTemplates.Where{
+        $_.Subject -match $Subject -and
+        $_.Extensions.Where{
+            $_.Oid.FriendlyName -eq 'Certificate Template Name'
+        }.Format($false) -eq $TemplateName
+    } | Sort-Object -Property 'NotAfter' -Descending | Select-Object -First 1
+
```

Comment:
If more than one cert is found which satisfy the criteria. I think It should error out instead of choosing one based on the expiry date. The reason being, the same cert have to be trusted by all clients connecting to this server and we can't tell which cert is trusted by all clients connecting to this PullServer.

Reply:
Yes, there is a requirement that the user explicitly say that they intended for the endpoint to `AllowUnencryptedTraffic`